src/m365/spfx/commands/project/project-github-workflow-add.ts

```diff
@@ -0,0 +1,203 @@
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      throw new CommandError(`Couldn't find project root folder`, SpfxProjectGithubWorkflowAddCommand.ERROR_NO_PROJECT_ROOT_FOLDER);
+    }
+
+    const solution: string = path.join(this.projectRootPath as string, 'sharepoint', 'solution');
+    if (!fs.existsSync(solution)) {
+      throw new CommandError(`No SPFx app package found`);
+    }
+
+    if (this.debug) {
+      logger.logToStderr(`Adding GitHub workflow in the current SPFx project`);
+    }
+
+    try {
+      this.updateWorkflow(solution, workflow, args.options);
+      this.saveWorkflow(workflow);
+    }
+    catch (error: any) {
+      throw new CommandError(error);
+    }
+  }
+
+  private saveWorkflow(workflow: any): void {
+    const githubPath: string = path.join(this.projectRootPath as string, '.github');
+    this.ensureDirectory(githubPath);
+
+    const workflowPath: string = path.join(githubPath, 'workflows');
+    this.ensureDirectory(workflowPath);
+
+    const workflowFile: string = path.join(workflowPath, 'deploy-spfx-solution.yml');
+    fs.writeFileSync(path.resolve(workflowFile), yaml.stringify(workflow), 'utf-8');
+  }
+
+  private updateWorkflow(solution: string, workflow: any, options: GlobalOptions): void {
+    let sppkgFileName: string = fs.readdirSync(solution).filter(fn => fn.endsWith('.sppkg'))[0];
+    sppkgFileName = sppkgFileName.replace('.sppkg', '');
+
+    workflow.name = workflow.name.replace('{{ name }}', options.name ?? sppkgFileName);
+
+    if (options.branchName) {
+      workflow.on.push.branches[0] = options.branchName;
+    }
+
+    if (options.manuallyTrigger) {
+      workflow.on['workflow_dispatch'] = null;
+    }
+
+    if (options.skipFeatureDeployment) {
+      this.getDeployAction(workflow).with['SKIP_FEATURE_DEPLOYMENT'] = true;
+    }
+
+    if (options.overwrite) {
+      this.getDeployAction(workflow).with['OVERWRITE'] = true;
+    }
+
+    if (options.loginMethod === 'user') {
+      const loginAction = this.getLoginAction(workflow);
+      loginAction.with = {};
+      loginAction.with['ADMIN_USERNAME'] = '${{ secrets.ADMIN_USERNAME }}';
+      loginAction.with['ADMIN_PASSWORD'] = '${{ secrets.ADMIN_PASSWORD }}';
+    }
+
+    if (options.scope === 'sitecollection') {
+      const deployAction = this.getDeployAction(workflow);
+      deployAction.with['SCOPE'] = 'sitecollection';
+      deployAction.with['SITE_COLLECTION_URL'] = options.siteUrl;
+    }
+
+    if (sppkgFileName) {
+      const deployAction = this.getDeployAction(workflow);
+      deployAction.with['APP_FILE_PATH'] = deployAction.with['APP_FILE_PATH'].replace('{{ solutionName }}', sppkgFileName);
+    }
+  }
+
+  private getLoginAction(workflow: any): any {
+    const steps = this.getWorkFlowSteps(workflow);
+    return steps[steps.length - 2];
```

Comment:
Let's look for a more reliable way of looking up the relevant parts of the definition. This makes an implicit assumption on the current structure and if we change it and forgot to change this part, which we totally could, we'll get wrong results.